### ISSUE TITLE
0.20.1  Still getting error on migration from h2 to mysql, PermissionsGroup Duplicate Entry

### ISSUE THREAD

**Issue by kevinjmullen** [metabase/metabase#3637]
metabase 0.20.1 on an ubuntu aws instance

Is there a set of commands I can run to clear this?

`Transfering 3 instances of PermissionsGroup....BatchUpdateException:
 Message: Duplicate entry '3' for key 'PRIMARY'
 SQLState: 23000
 Error Code: 1062
java.sql.BatchUpdateException: Duplicate entry '3' for key 'PRIMARY'
	at com.mysql.jdbc.SQLError.createBatchUpdateException(SQLError.java:1166)
	at com.mysql.jdbc.PreparedStatement.executeBatchSerially(PreparedStatement.java:1773)
	at com.mysql.jdbc.PreparedStatement.executeBatchInternal(PreparedStatement.java:1257)
	at com.mysql.jdbc.StatementImpl.executeBatch(StatementImpl.java:958)
	at clojure.java.jdbc$execute_batch.invokeStatic(jdbc.clj:426)
	at clojure.java.jdbc$execute_batch.invoke(jdbc.clj:419)
	at clojure.java.jdbc$db_do_execute_prepared_statement$fn__2749.invoke(jdbc.clj:792)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:637)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:598)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invokeStatic(jdbc.clj:791)
	at clojure.java.jdbc$db_do_execute_prepared_statement.invoke(jdbc.clj:778)
	at clojure.java.jdbc$db_do_prepared.invokeStatic(jdbc.clj:814)
	at clojure.java.jdbc$db_do_prepared.invoke(jdbc.clj:795)
	at clojure.java.jdbc$insert_cols_BANG_.invokeStatic(jdbc.clj:1069)
	at clojure.java.jdbc$insert_cols_BANG_.invoke(jdbc.clj:1061)
	at clojure.java.jdbc$insert_multi_BANG_.invokeStatic(jdbc.clj:1110)
	at clojure.java.jdbc$insert_multi_BANG_.invoke(jdbc.clj:1093)
	at metabase.cmd.load_from_h2$insert_entity_BANG_$fn__31010.invoke(load_from_h2.clj:125)
	at metabase.cmd.load_from_h2$insert_entity_BANG_.invokeStatic(load_from_h2.clj:124)
	at metabase.cmd.load_from_h2$insert_entity_BANG_.invoke(load_from_h2.clj:106)
	at metabase.cmd.load_from_h2$load_data_BANG_.invokeStatic(load_from_h2.clj:140)
	at metabase.cmd.load_from_h2$load_data_BANG_.invoke(load_from_h2.clj:134)
	at metabase.cmd.load_from_h2$load_from_h2_BANG_$fn__31090.invoke(load_from_h2.clj:209)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:613)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:629)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:598)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at metabase.cmd.load_from_h2$load_from_h2_BANG_.invokeStatic(load_from_h2.clj:206)
	at metabase.cmd.load_from_h2$load_from_h2_BANG_.invoke(load_from_h2.clj:199)
	at clojure.lang.Var.invoke(Var.java:379)
	at metabase.core$load_from_h2.invokeStatic(core.clj:218)
	at metabase.core$load_from_h2.invoke(core.clj:211)
	at metabase.core$load_from_h2.invokeStatic(core.clj:214)
	at metabase.core$load_from_h2.invoke(core.clj:211)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.AFn.applyTo(AFn.java:144)
	at clojure.core$apply.invokeStatic(core.clj:646)
	at clojure.core$apply.invoke(core.clj:641)
	at metabase.core$run_cmd$fn__31843.invoke(core.clj:265)
	at metabase.core$run_cmd.invokeStatic(core.clj:265)
	at metabase.core$run_cmd.doInvoke(core.clj:264)
	at clojure.lang.RestFn.invoke(RestFn.java:410)
	at clojure.lang.AFn.applyToHelper(AFn.java:154)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at clojure.core$apply.invokeStatic(core.clj:648)
	at clojure.core$apply.invoke(core.clj:641)
	at metabase.core$_main.invokeStatic(core.clj:279)
	at metabase.core$_main.doInvoke(core.clj:275)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at metabase.core.main(Unknown Source)
Caused by: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry '3' for key 'PRIMARY'
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:57)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)
	at com.mysql.jdbc.Util.handleNewInstance(Util.java:404)
	at com.mysql.jdbc.Util.getInstance(Util.java:387)
	at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:934)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3966)
	at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3902)
	at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2526)
	at com.mysql.jdbc.MysqlIO.sqlQueryDirect(MysqlIO.java:2673)
	at com.mysql.jdbc.ConnectionImpl.execSQL(ConnectionImpl.java:2549)
	at com.mysql.jdbc.PreparedStatement.executeInternal(PreparedStatement.java:1861)
	at com.mysql.jdbc.PreparedStatement.executeUpdateInternal(PreparedStatement.java:2073)
	at com.mysql.jdbc.PreparedStatement.executeBatchSerially(PreparedStatement.java:1751)
	... 52 more
Command failed with exception: Duplicate entry '3' for key 'PRIMARY'`

**Comment by arabold** [comment]
Similar problem here.

I've updated to 0.20.1 and it works smooth. Then I wanted to migrade from H2 to Postgres. 

First I ran the migration: `java -jar metabase.jar load-from-h2 /opt/metabase/metabase.db`
```
10-20 17:24:33 INFO metabase.util :: Loading Metabase...
10-20 17:24:51 INFO metabase.db :: Verifying postgres Database Connection ...
10-20 17:24:51 DEBUG metabase.driver :: Registered driver :postgres 🚚
10-20 17:24:51 INFO metabase.db :: Verify Database Connection ... ✅
10-20 17:24:51 INFO metabase.db :: Running Database Migrations...
10-20 17:24:51 INFO metabase.db :: Setting up Liquibase...
10-20 17:24:52 INFO metabase.db :: Liquibase is ready.
10-20 17:24:52 INFO metabase.db :: Checking if Database has unran migrations...
10-20 17:26:04 INFO metabase.db :: Database has unran migrations. Waiting for migration lock to be cleared...
10-20 17:26:04 INFO metabase.db :: Migration lock is cleared. Running migrations...
10-20 17:26:57 INFO metabase.db :: Database Migrations Current ... ✅
com.mchange.v2.cfg.DelayedLogItem [ level -> FINE, text -> "The configuration file for resource identifier 'hocon:/reference,/application,/c3p0,/' could not be found. Skipping.", exception -> null]
Temporarily disabling DB constraints...
[OK]
Transfering 4 instances of Database....[OK]
Transfering 315 instances of RawTable.....[OK]
Transfering 9802 instances of RawColumn....................................[OK]
Transfering 6 instances of User....[OK]
Transfering 14 instances of Setting....[OK]
Transfering 315 instances of Table.....[OK]
Transfering 9802 instances of Field....................................[OK]
Transfering 1334 instances of FieldValues........[OK]
Transfering 1 instances of Segment....[OK]
Transfering 163 instances of Revision....[OK]
Transfering 738 instances of ViewLog......[OK]
Transfering 38 instances of Session....[OK]
Transfering 4 instances of Dashboard....[OK]
Transfering 39 instances of Card....[OK]
Transfering 11 instances of CardFavorite....[OK]
Transfering 18 instances of DashboardCard....[OK]
Transfering 5 instances of DashboardCardSeries....[OK]
Transfering 163 instances of Activity....[OK]
Transfering 5008 instances of QueryExecution....................[OK]
Transfering 2 instances of Pulse....[OK]
Transfering 10 instances of PulseCard....[OK]
Transfering 2 instances of PulseChannel....[OK]
Transfering 7 instances of PulseChannelRecipient....[OK]
Transfering 3 instances of Label....[OK]
Transfering 22 instances of CardLabel....[OK]
Transfering 3 instances of PermissionsGroup....[OK]
Transfering 9 instances of PermissionsGroupMembership....[OK]
Transfering 9 instances of Permissions....[OK]
Setting postgres sequence ids to proper values...
[OK]
```

Then I start up `java -jar metabase.jar` which fails with the following error:
```
10-20 17:28:01 DEBUG metabase.driver :: Registered driver :postgres 🚚
10-20 17:28:01 DEBUG metabase.driver :: Registered driver :redshift 🚚
10-20 17:28:01 DEBUG metabase.driver :: Registered driver :sqlite 🚚
10-20 17:28:01 DEBUG metabase.driver :: Registered driver :sqlserver 🚚
10-20 17:28:01 INFO metabase.db :: Verifying postgres Database Connection ...
10-20 17:28:01 INFO metabase.db :: Verify Database Connection ... ✅
10-20 17:28:01 INFO metabase.db :: Running Database Migrations...
10-20 17:28:01 INFO metabase.db :: Setting up Liquibase...
10-20 17:28:02 INFO metabase.db :: Liquibase is ready.
10-20 17:28:02 INFO metabase.db :: Checking if Database has unran migrations...
10-20 17:29:14 INFO metabase.db :: Database Migrations Current ... ✅
com.mchange.v2.cfg.DelayedLogItem [ level -> FINE, text -> "The configuration file for resource identifier 'hocon:/reference,/application,/c3p0,/' could not be found. Skipping.", exception -> null]
10-20 17:29:15 INFO db.migrations :: Running all necessary data migrations, this may take a minute.
10-20 17:29:15 INFO db.migrations :: Running data migration 'set-card-database-and-table-ids'...
10-20 17:29:15 INFO db.migrations :: Running data migration 'set-mongodb-databases-ssl-false'...
10-20 17:29:15 INFO db.migrations :: Running data migration 'set-default-schemas'...
10-20 17:29:15 INFO db.migrations :: Running data migration 'set-admin-email'...
10-20 17:29:15 INFO db.migrations :: Running data migration 'remove-database-sync-activity-entries'...
10-20 17:29:15 INFO db.migrations :: Running data migration 'update-dashboards-to-new-grid'...
10-20 17:29:15 INFO db.migrations :: Running data migration 'migrate-field-visibility-type'...
10-20 17:29:15 INFO db.migrations :: Running data migration 'fix-dashboard-cards-without-positions'...
10-20 17:29:15 INFO db.migrations :: Running data migration 'create-raw-tables'...
10-20 17:29:15 INFO db.migrations :: Running data migration 'add-users-to-default-permissions-groups'...
org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "unique_permissions_group_membership_user_id_group_id"
  Detail: Key (user_id, group_id)=(37, 1) already exists.
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2198)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:1927)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:255)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.execute(AbstractJdbc2Statement.java:561)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeWithFlags(AbstractJdbc2Statement.java:419)
	at org.postgresql.jdbc2.AbstractJdbc2Statement.executeUpdate(AbstractJdbc2Statement.java:365)
	at com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeUpdate(NewProxyPreparedStatement.java:105)
	at clojure.java.jdbc$db_do_execute_prepared_return_keys$exec_and_return_keys__2732.invoke(jdbc.clj:733)
	at clojure.java.jdbc$db_do_execute_prepared_return_keys.invokeStatic(jdbc.clj:747)
	at clojure.java.jdbc$db_do_execute_prepared_return_keys.invoke(jdbc.clj:726)
	at clojure.java.jdbc$db_do_prepared_return_keys.invokeStatic(jdbc.clj:774)
	at clojure.java.jdbc$db_do_prepared_return_keys.invoke(jdbc.clj:755)
	at clojure.java.jdbc$multi_insert_helper$fn__2796.invoke(jdbc.clj:996)
	at clojure.core$map$fn__4785.invoke(core.clj:2644)
	at clojure.lang.LazySeq.sval(LazySeq.java:40)
	at clojure.lang.LazySeq.seq(LazySeq.java:49)
	at clojure.lang.RT.seq(RT.java:521)
	at clojure.core$seq__4357.invokeStatic(core.clj:137)
	at clojure.core$dorun.invokeStatic(core.clj:3024)
	at clojure.core$doall.invokeStatic(core.clj:3039)
	at clojure.core$doall.invoke(core.clj:3039)
	at clojure.java.jdbc$multi_insert_helper.invokeStatic(jdbc.clj:996)
	at clojure.java.jdbc$multi_insert_helper.invoke(jdbc.clj:991)
	at clojure.java.jdbc$insert_helper$fn__2799.invoke(jdbc.clj:1003)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:613)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:598)
	at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:585)
	at clojure.java.jdbc$insert_helper.invokeStatic(jdbc.clj:1003)
	at clojure.java.jdbc$insert_helper.invoke(jdbc.clj:998)
	at clojure.java.jdbc$insert_rows_BANG_.invokeStatic(jdbc.clj:1059)
	at clojure.java.jdbc$insert_rows_BANG_.invoke(jdbc.clj:1047)
	at clojure.java.jdbc$insert_multi_BANG_.invokeStatic(jdbc.clj:1109)
	at clojure.java.jdbc$insert_multi_BANG_.invoke(jdbc.clj:1093)
	at metabase.db$simple_insert_many_BANG_.invokeStatic(db.clj:733)
	at metabase.db$simple_insert_many_BANG_.invoke(db.clj:720)
	at metabase.db$simple_insert_BANG_.invokeStatic(db.clj:758)
	at metabase.db$simple_insert_BANG_.invoke(db.clj:748)
	at metabase.db$insert_BANG_.invokeStatic(db.clj:773)
	at metabase.db$insert_BANG_.invoke(db.clj:762)
	at metabase.db$insert_BANG_.invokeStatic(db.clj:776)
	at metabase.db$insert_BANG_.doInvoke(db.clj:762)
	at clojure.lang.RestFn.invoke(RestFn.java:494)
	at metabase.db.migrations$add_users_to_default_permissions_groups.invokeStatic(migrations.clj:231)
	at metabase.db.migrations$add_users_to_default_permissions_groups.invoke(migrations.clj:226)
	at metabase.db.migrations$run_migration_if_needed_BANG_.invokeStatic(migrations.clj:39)
	at metabase.db.migrations$run_migration_if_needed_BANG_.invoke(migrations.clj:30)
	at metabase.db.migrations$run_all.invokeStatic(migrations.clj:59)
	at metabase.db.migrations$run_all.invoke(migrations.clj:53)
	at clojure.lang.Var.invoke(Var.java:375)
	at metabase.db$setup_db_BANG_.invokeStatic(db.clj:381)
	at metabase.db$setup_db_BANG_.doInvoke(db.clj:349)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at metabase.core$init_BANG_.invokeStatic(core.clj:112)
	at metabase.core$init_BANG_.invoke(core.clj:92)
	at metabase.core$start_normally.invokeStatic(core.clj:195)
	at metabase.core$start_normally.invoke(core.clj:189)
	at metabase.core$_main.invokeStatic(core.clj:280)
	at metabase.core$_main.doInvoke(core.clj:275)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.AFn.applyToHelper(AFn.java:152)
	at clojure.lang.RestFn.applyTo(RestFn.java:132)
	at metabase.core.main(Unknown Source)
10-20 17:29:16 ERROR metabase.core :: Metabase Initialization FAILED:  ERROR: duplicate key value violates unique constraint "unique_permissions_group_membership_user_id_group_id"
  Detail: Key (user_id, group_id)=(37, 1) already exists.
10-20 17:29:16 INFO metabase.core :: Metabase Shutting Down ...
10-20 17:29:16 INFO metabase.core :: Metabase Shutdown COMPLETE
```

**Comment by camsaul** [comment]
@kevinjmullen @arabold sorry for this obnoxious bug. You can fix it manually with v0.20.1 by connecting to your DB and running this SQL:

```sql
INSERT INTO data_migrations (id, "timestamp") 
VALUES ('add-users-to-default-permissions-groups', now());
```

(for Postgres). 

I've fixed this issue in #3646

**Comment by camsaul** [comment]
@kevinjmullen @arabold we're planning on shipping a bugfix release later today with the fix (#3646)

**Comment by camsaul** [comment]
@kevinjmullen @arabold we're planning on shipping a bugfix in the next few days with a fix (#3646) 

In the meantime you can fix this manually by connecting to your database and executing this SQL (for Postgres):

```sql
INSERT INTO data_migrations (id, "timestamp") 
VALUES ('add-users-to-default-permissions-groups', now());
```

**Comment by camsaul** [comment]
This was fixed by #3646
